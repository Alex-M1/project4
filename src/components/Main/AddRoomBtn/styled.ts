import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';

export const StBtn = styled.button <IStyled>`
    position: absolute;
    top: 30px;
    right: 35px;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    box-shadow: 0 0 4px #000;
    border: none;
    background: ${({ colors, theme }) => colors[theme].button};
    &:hover {
        box-shadow: 0 0 5px ${({ colors, theme }) => colors[theme].AddRoomBtn};;
    }
    &::before {
        content: '';
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        width: 15px;
        height: 3px;
        background: ${({ colors, theme }) => colors[theme].AddRoomBtn};;
    }
    &::after {
        content: '';
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        width: 3px;
        height: 15px;
        background: ${({ colors, theme }) => colors[theme].AddRoomBtn};;
    }
`;
