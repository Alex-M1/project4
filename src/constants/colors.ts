export const colors = {
  light: {
    background: '#fff',
    backgroundSecondary: '#ECECEC',
    headerBg: '#2B5278',
    backgroundItem: '#d7d8e2',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#D1D1D1',
    textColor: '#000',
    textColorHeader: '#fff',
    placeholderColor: '#55667B',
    AddRoomBtn: '#fff',
    AddRoomBtnHover: '#fff',
    backgroundModal: 'rgba(0,0,0, 0.8)',
    backgroundModalShadow: 'rgba(0,0,0, 0.8)',
    backgroundModalContainer: '#dcd9d9',
    selectBackground: '#d4d4d4',
    iconTheme: '#ff0',
    iconThemeMoon: 'rgba(0,0,0,0)',
    iconThemeSun: '#ff0',
    ticTacColor: '#000',
    whiteChecker: '#fff',
    blackChecker: '#777',
    gray: '#ccc',
    white: '#fff',
    black: '#000',
    blackCellBg: '#777',
    hoverLight: '#ddd',
    gold: '#FFD700',
  },
  dark: {
    background: '#0E1621',
    backgroundSecondary: '#17212B',
    headerBg: '#2B5278',
    backgroundItem: '#25323f',
    button: '#2B5278',
    buttonColor: '#fff',
    buttonHover: '#346392',
    input: '#242F3D',
    textColor: '#d4d4d4',
    textColorHeader: '#d4d4d4',
    placeholderColor: '#55667B',
    AddRoomBtn: '#fff',
    AddRoomBtnHover: '#fff',
    backgroundModal: 'rgba(0,0,0, 0.8)',
    backgroundModalShadow: '#757575',
    backgroundModalContainer: '#17212B',
    selectBackground: '#d4d4d4',
    iconTheme: '#fff',
    iconThemeMoon: '#2B5278',
    iconThemeSun: 'rgba(0,0,0,0)',
    ticTacColor: '#fff',
    whiteChecker: '#fff',
    blackChecker: '#777',
    gray: '#ccc',
    white: '#fff',
    black: '#000',
    blackCellBg: '#777',
    hoverLight: '#ddd',
    gold: '#FFD700',
  },
};
