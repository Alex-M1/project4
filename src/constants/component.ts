export const CELL = [
  {
    id: 1,
    color: 'black',
  },
  {
    id: 2,
    color: 'white',
  },
  {
    id: 3,
    color: 'black',
  },
  {
    id: 4,
    color: 'white',
  },
  {
    id: 5,
    color: 'black',
  },
  {
    id: 6,
    color: 'white',
  },
  {
    id: 7,
    color: 'black',
  },
  {
    id: 8,
    color: 'white',
  },
  {
    id: 9,
    color: 'black',
  },
  {
    id: 10,
    color: 'white',
  },
  {
    id: 11,
    color: 'black',
  },
  {
    id: 12,
    color: 'white',
  },
  {
    id: 13,
    color: 'black',
  },
  {
    id: 14,
    color: 'white',
  },
  {
    id: 15,
    color: 'black',
  },
  {
    id: 16,
    color: 'white',
  },
  {
    id: 17,
    color: 'black',
  },
  {
    id: 18,
    color: 'white',
  },
  {
    id: 19,
    color: 'black',
  },
  {
    id: 20,
    color: 'white',
  },
  {
    id: 21,
    color: 'black',
  },
  {
    id: 22,
    color: 'white',
  },
  {
    id: 23,
    color: 'black',
  },
  {
    id: 24,
    color: 'white',
  },
  {
    id: 25,
    color: 'black',
  },
  {
    id: 26,
    color: 'white',
  },
  {
    id: 27,
    color: 'black',
  },
  {
    id: 28,
    color: 'white',
  },
  {
    id: 29,
    color: 'black',
  },
  {
    id: 30,
    color: 'white',
  },
  {
    id: 31,
    color: 'black',
  },
  {
    id: 32,
    color: 'white',
  },
  {
    id: 33,
    color: 'black',
  },
  {
    id: 34,
    color: 'white',
  },
  {
    id: 35,
    color: 'black',
  },
  {
    id: 36,
    color: 'white',
  },
  {
    id: 37,
    color: 'black',
  },
  {
    id: 38,
    color: 'white',
  },
  {
    id: 39,
    color: 'black',
  },
  {
    id: 40,
    color: 'white',
  },
  {
    id: 41,
    color: 'black',
  },
  {
    id: 42,
    color: 'white',
  },
  {
    id: 43,
    color: 'black',
  },
  {
    id: 44,
    color: 'white',
  },
  {
    id: 45,
    color: 'black',
  },
  {
    id: 46,
    color: 'white',
  },
  {
    id: 47,
    color: 'black',
  },
  {
    id: 48,
    color: 'white',
  },
  {
    id: 49,
    color: 'black',
  },
  {
    id: 50,
    color: 'white',
  },
  {
    id: 51,
    color: 'black',
  },
  {
    id: 52,
    color: 'white',
  },
  {
    id: 53,
    color: 'black',
  },
  {
    id: 54,
    color: 'white',
  },
  {
    id: 55,
    color: 'black',
  },
  {
    id: 56,
    color: 'white',
  },
  {
    id: 57,
    color: 'black',
  },
  {
    id: 58,
    color: 'white',
  },
  {
    id: 59,
    color: 'black',
  },
  {
    id: 60,
    color: 'white',
  },
  {
    id: 61,
    color: 'black',
  },
  {
    id: 62,
    color: 'white',
  },
  {
    id: 63,
    color: 'black',
  },
  {
    id: 64,
    color: 'white',
  },
];
