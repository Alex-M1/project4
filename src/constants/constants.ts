export const MESSAGES = {
  invalid_login: 'invalid_login',
  invalid_password: 'invalid_password',
  invalid_confirm: 'invalid_confirm',
};

export const REG_EXP = {
  login: /^[A-Za-z0-9]{1,25}/,
  password: /^[A-Za-z0-9]{6,25}$/,
};

export const COOKIE_SETTING = {
  AGE: 20000,
  SPLIT_LEN: 2,
};

export const TIC_TAC = {
  WINNER_LINES: [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
    [0, 3, 6],
    [1, 4, 7],
    [2, 5, 8],
    [0, 4, 8],
    [2, 4, 6],
  ],
  FIELD_SIZE: 480,
};
export const COOKIE_AGE = 20000;

export const GAME_TYPE = {
  TIC_TAC_TOE: 'Tic-tac-toe',
  CHECKERS: 'Checkers',
};

export const TIC_TAC_ITEM = {
  X: 'X',
  O: 'O',
};

export const CHESS_DESK = {
  COLS: [1, 2, 3, 4, 5, 6, 7, 8],
  ROWS: [1, 2, 3, 4, 5, 6, 7, 8],
  CELLS_NUM_INIT: 0,
};

export const LANGUAGE = {
  LANG: 'lang',
  EN: 'en',
  RU: 'ru',
};

export const VIEW_OPTIONS = {
  HEADER_HEIGHT: 90,
  DRAGGIN_OPACITY: {
    IS_DRAGGING: 0.3,
    NOT_DRAGGING: 1,
  },
  CHECKER_FIELD: 500,
  CELL_BACKGROUND: {
    GRAY: '#ccc',
    WHITE: '#fff',
    POSSIBLE: '#b2ff65',
  },
};

export const LOCAL_STORAGE = {
  login: 'login',
  token: 'token',
  gameOptions: 'gameOptions',
};

export const GAME_SETTINGS = {
  bot: 'Bot',
  user: 'User',
};

export const CHECKER_FIELD_INIT = [
  null,
  {
    checker: { blackChecker: false, queen: false },
    index: 1,
    blackSquare: true,
  },
  {
    checker: null,
    index: 2,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 3,
    blackSquare: true,
  },
  {
    checker: null, index: 4, blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 5,
    blackSquare: true,
  },
  {
    checker: null,
    index: 6,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 7,
    blackSquare: true,
  },
  { checker: null, index: 8, blackSquare: false },
  { checker: null, index: 9, blackSquare: false },
  {
    checker: { blackChecker: false, queen: false },
    index: 10,
    blackSquare: true,
  },
  {
    checker: null,
    index: 11,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 12,
    blackSquare: true,
  },
  {
    checker: null,
    index: 13,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 14,
    blackSquare: true,
  },
  {
    checker: null,
    index: 15,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 16,
    blackSquare: true,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 17,
    blackSquare: true,
  },
  {
    checker: null,
    index: 18,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 19,
    blackSquare: true,
  }, {
    checker: null,
    index: 20,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 21,
    blackSquare: true,
  },
  {
    checker: null,
    index: 22,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: false,
      queen: false,
    },
    index: 23,
    blackSquare: true,
  },
  {
    checker: null,
    index: 24,
    blackSquare: false,
  },
  {
    checker: null,
    index: 25,
    blackSquare: false,
  },
  {
    checker: null,
    index: 26,
    blackSquare: true,
  },
  {
    checker: null,
    index: 27,
    blackSquare: false,
  },
  {
    checker: null,
    index: 28,
    blackSquare: true,
  },
  {
    checker: null,
    index: 29,
    blackSquare: false,
  },
  {
    checker: null,
    index: 30,
    blackSquare: true,
  },
  {
    checker: null,
    index: 31,
    blackSquare: false,
  },
  {
    checker: null,
    index: 32,
    blackSquare: true,
  },
  {
    checker: null,
    index: 33,
    blackSquare: true,
  },
  {
    checker: null,
    index: 34,
    blackSquare: false,
  },
  {
    checker: null,
    index: 35,
    blackSquare: true,
  },
  {
    checker: null,
    index: 36,
    blackSquare: false,
  },
  {
    checker: null,
    index: 37,
    blackSquare: true,
  },
  {
    checker: null,
    index: 38,
    blackSquare: false,
  },
  {
    checker: null,
    index: 39,
    blackSquare: true,
  },
  {
    checker: null,
    index: 40,
    blackSquare: false,
  },
  {
    checker: null,
    index: 41,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 42,
    blackSquare: true,
  },
  {
    checker: null,
    index: 43,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 44,
    blackSquare: true,
  },
  {
    checker: null,
    index: 45,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 46,
    blackSquare: true,
  },
  {
    checker: null,
    index: 47,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 48,
    blackSquare: true,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 49,
    blackSquare: true,
  },
  {
    checker: null,
    index: 50,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 51,
    blackSquare: true,
  },
  {
    checker: null,
    index: 52,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 53,
    blackSquare: true,
  },
  {
    checker: null,
    index: 54,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 55,
    blackSquare: true,
  },
  {
    checker: null,
    index: 56,
    blackSquare: false,
  },
  {
    checker: null,
    index: 57,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 58,
    blackSquare: true,
  },
  {
    checker: null,
    index: 59,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 60,
    blackSquare: true,
  },
  {
    checker: null,
    index: 61,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 62,
    blackSquare: true,
  },
  {
    checker: null,
    index: 63,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 64,
    blackSquare: true,
  },
];
