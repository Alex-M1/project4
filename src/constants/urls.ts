export const url = {
  authClient: '/',
  regClient: '/registration',
  server: 'http://35.176.167.155:8089',
  registration: '/registration/reg',
  auth: '/authorization/auth',
};
