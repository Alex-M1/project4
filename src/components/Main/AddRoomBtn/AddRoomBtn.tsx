import React, { useState } from 'react';
import { useTheme } from 'src/components/hooks/useTheme';
import Modal from 'src/components/Modal';
import { StBtn } from './styled';

const AddRoomBtn: React.FC = () => {
    const { colors, theme } = useTheme();
    const [isOpen, setIsOpen] = useState(false);
    const handleClick = () => setIsOpen((prev) => !prev);

    return (
        <>
            <StBtn
                theme={theme}
                colors={colors}
                onClick={handleClick}
            />
            {isOpen && <Modal handleCloseModal={handleClick} />}
        </>
    );
};

export default AddRoomBtn;
