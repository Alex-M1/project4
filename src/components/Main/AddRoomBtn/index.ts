import AddRoomBtn from './AddRoomBtn';

export default AddRoomBtn;
