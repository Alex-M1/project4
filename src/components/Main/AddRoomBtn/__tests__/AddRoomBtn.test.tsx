import { mount, shallow } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import { act } from 'react-dom/test-utils';
import { GAME_TYPE } from 'constants/constants';
import { mountSmart } from 'src/__tests__/testHelper';
import AddRoomBtn from '../AddRoomBtn';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);
const mockStore = configureStore();
const store = mockStore({
  room: {
    rooms: [],
    gameType: GAME_TYPE.CHECKERS,
    toRoom: '',
  },
});
describe('AddRoomBtn', () => {
  let portal;
  beforeEach(() => {
    portal = document.createElement('div');
    portal.setAttribute('id', 'portal');
    document.body.append(portal);
  });
  it('Should match snapshot', () => {
    const component = shallow(<AddRoomBtn />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StBtn', () => {
    const component = mount(<AddRoomBtn />);
    expect(component.find('styled__StBtn')).toHaveLength(1);
  });
  it('should not render Modal', () => {
    const component = mount(<AddRoomBtn />);
    expect(component.find('Modal')).not.toHaveLength(1);
  });
  it('should render Modal', () => {
    const component = mountSmart(<AddRoomBtn />, store);
    act(() => component.find('styled__StBtn').props().onClick());
    component.update();
    expect(component.find('Modal')).toHaveLength(1);
  });
});
