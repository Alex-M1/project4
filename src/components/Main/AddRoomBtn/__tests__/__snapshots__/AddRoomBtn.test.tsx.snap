// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRoomBtn Should match snapshot 1`] = `"<button class=\\"styled__StBtn-kAydTA jyFLiv\\"></button>"`;
